### PR TITLE
[1LP][RFR] ansible deployment test for pod appliance

### PR DESCRIPTION
**Pull request by izapolsk**
REQUIRES: https://github.com/ManageIQ/wrapanapi/pull/322

{{ pytest: cfme/tests/pod/test_appliance_crud.py --use-provider cm-pod1  }}

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit fe52502bd0439756d089fea226715fb9a1a9bd02:

`cfme/tests/pod/test_appliance_crud.py`:
- :heavy_exclamation_mark: Line 32:73: **W291** *trailing whitespace*
- :heavy_exclamation_mark: Line 34:64: **W291** *trailing whitespace*
- :heavy_exclamation_mark: Line 35:55: **W291** *trailing whitespace*
- :heavy_exclamation_mark: Line 36:59: **W291** *trailing whitespace*
- :heavy_exclamation_mark: Line 37:82: **W291** *trailing whitespace*
- :heavy_exclamation_mark: Line 50:92: **W291** *trailing whitespace*
- :heavy_exclamation_mark: Line 52:87: **W291** *trailing whitespace*
- :heavy_exclamation_mark: Line 53:84: **W291** *trailing whitespace*
- :heavy_exclamation_mark: Line 54:79: **W291** *trailing whitespace*
- :heavy_exclamation_mark: Line 55:78: **W291** *trailing whitespace*
- :heavy_exclamation_mark: Line 58:83: **W291** *trailing whitespace*
- :heavy_exclamation_mark: Line 59:82: **W291** *trailing whitespace*
- :heavy_exclamation_mark: Line 60:77: **W291** *trailing whitespace*
- :bangbang: Line 73:101: **E501** *line too long (102 > 100 characters)*
- :red_circle: Line 122:5: **F841** *local variable 'ansible_cmd' is assigned to but never used*

Hmmm
:scream: You might like to check the lint of your code :point_up: 


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit e0d3ecc344fd3082246a49f629f47003191a1f87:

`cfme/tests/pod/test_appliance_crud.py`:
- :heavy_exclamation_mark: Line 32:73: **W291** *trailing whitespace*
- :heavy_exclamation_mark: Line 34:64: **W291** *trailing whitespace*
- :heavy_exclamation_mark: Line 35:55: **W291** *trailing whitespace*
- :heavy_exclamation_mark: Line 36:59: **W291** *trailing whitespace*
- :heavy_exclamation_mark: Line 37:82: **W291** *trailing whitespace*
- :heavy_exclamation_mark: Line 50:92: **W291** *trailing whitespace*
- :heavy_exclamation_mark: Line 52:87: **W291** *trailing whitespace*
- :heavy_exclamation_mark: Line 53:84: **W291** *trailing whitespace*
- :heavy_exclamation_mark: Line 54:79: **W291** *trailing whitespace*
- :heavy_exclamation_mark: Line 55:78: **W291** *trailing whitespace*
- :heavy_exclamation_mark: Line 58:83: **W291** *trailing whitespace*
- :heavy_exclamation_mark: Line 59:82: **W291** *trailing whitespace*
- :heavy_exclamation_mark: Line 60:77: **W291** *trailing whitespace*
- :bangbang: Line 73:101: **E501** *line too long (102 > 100 characters)*
- :red_circle: Line 122:5: **F841** *local variable 'ansible_cmd' is assigned to but never used*

Hmmm
:scream: You might like to check the lint of your code :point_up: 


**No commit flaws detected**.

*CFME QE Bot*

**Comment by mshriver** [comment]
wrapanapi req filled by #7598

File: cfme/tests/pod/test_appliance_crud.py [hunk 1]
```diff
@@ -13,29 +18,153 @@
...
+
+@pytest.fixture
+def template_tags(appliance):
+    api = trackerbot.api()
+    stream = get_stream(appliance.version.vstring)
+    template_data = trackerbot.latest_template(api, stream)
+    template = api.template(template_data['latest_template']).get()
+    # TODO: prepare some serialization functions and add them everywhere
+    tags = json.loads(template['custom_data'].replace("'", '"').replace('u"', '"'))
+    return tags['TAGS']
```
Comment (mshriver) [inline cfme/tests/pod/test_appliance_crud.py]: Consider wrapping this in a skip if trackerbot isn't reachable, no stream parsed, there's no custom data, or there's no 'TAGS'

File: cfme/tests/pod/test_appliance_crud.py [hunk 2]
```diff
@@ -13,29 +18,153 @@
...
+{host}
+
+# host group for etcd
+[etcd]
+{host}
+
+# host group for nodes, includes region info
+[nodes]
+{host} openshift_node_labels="{{'region': 'infra', 'zone': 'default'}}" openshift_schedulable=true
+"""  # noqa
```
Comment (mshriver) [inline cfme/tests/pod/test_appliance_crud.py]: :tada:

File: cfme/tests/pod/test_appliance_crud.py [hunk 3]
```diff
@@ -13,29 +18,153 @@
...
+                app_tag=tags['cfme-openshift-app']['tag'],
+                ansible_url=tags['cfme-openshift-embedded-ansible']['url'],
+                ansible_tag=tags['cfme-openshift-embedded-ansible']['tag'],
+                httpd_url=tags['cfme-openshift-httpd']['url'],
+                httpd_tag=tags['cfme-openshift-httpd']['tag'],
+                memcached_url=tags['cfme-openshift-memcached']['url'],
+                memcached_tag=tags['cfme-openshift-memcached']['tag'],
+                db_url=tags['cfme-openshift-postgresql']['url'],
+                db_tag=tags['cfme-openshift-postgresql']['tag'])
+            with tempfile.NamedTemporaryFile('w') as f:
```
Comment (mshriver) [inline cfme/tests/pod/test_appliance_crud.py]: Consider logging the `fulfilled_config` so when reviewing results its easy to tell if it was formatted correctly.

File: cfme/tests/pod/test_appliance_crud.py [hunk 4]
```diff
@@ -13,29 +18,153 @@
...
+                f.write(fulfilled_config)
+                f.flush()
+                os.fsync(f.fileno())
+                remote_file = os.path.join('/tmp', f.name)
+                ssh_client.put_file(f.name, remote_file, ensure_host=True)
+
+                # run ansible deployment
+                ansible_cmd = ('/usr/bin/ansible-playbook -v -i {inventory_file} '
```
Comment (mshriver) [inline cfme/tests/pod/test_appliance_crud.py]: bump this out of the file-handle context as its not working with the file anymore.

File: cfme/tests/pod/test_appliance_crud.py [hunk 5]
```diff
@@ -13,29 +18,153 @@
...
+                               '/usr/share/ansible/openshift-ansible/playbooks/'
+                               'openshift-management/config.yml').format(inventory_file=remote_file)
+                cmd_result = ssh_client.run_command(ansible_cmd, ensure_host=True)
+                ssh_client.run_command('rm -f {f}'.format(f=remote_file))
+
+            assert cmd_result.success
```
Comment (mshriver) [inline cfme/tests/pod/test_appliance_crud.py]: Consider logging `cmd_result.output` so we know what ansible did.

File: cfme/tests/pod/test_appliance_crud.py (hunk 2 above)
Comment (izapolsk) [reply cfme/tests/pod/test_appliance_crud.py]: honestly, I was surprised.

File: cfme/tests/pod/test_appliance_crud.py [hunk 6]
```diff
@@ -13,29 +18,153 @@
...
+[masters]
+{host}
+
+# host group for etcd
+[etcd]
+{host}
+
+# host group for nodes, includes region info
+[nodes]
+{host} openshift_node_labels="{{'region': 'infra', 'zone': 'default'}}" openshift_schedulable=true
```
Comment (jawatts) [inline cfme/tests/pod/test_appliance_crud.py]: Sinlge master/node on one system, cool!

Comment (izapolsk) [reply cfme/tests/pod/test_appliance_crud.py]: hehe, I'd rather prefer to use one master and several nodes but it is what it is

File: cfme/tests/pod/test_appliance_crud.py [hunk 7]
```diff
@@ -13,29 +19,158 @@
...
+# SSH user, this user should allow ssh based auth without requiring a password
+ansible_ssh_user=root
+
+# If ansible_ssh_user is not root, ansible_become must be set to true
+ansible_become=true
+openshift_deployment_type=openshift-enterprise
+
+# uncomment the following to enable htpasswd authentication; defaults to
+#DenyAllPasswordIdentityProvider
+openshift_master_identity_providers=[{{'name': 'htpasswd_auth', 'login': 'true', 'challenge': 'true', 'kind': 'HTPasswdPasswordIdentityProvider', 'filename': '/etc/origin/master/htpasswd'}}]
```
Comment (jawatts) [inline cfme/tests/pod/test_appliance_crud.py]: I don't see you actually setting the login/password using htpasswd.

**Comment by izapolsk** [comment]
@mshriver, I made all requested changes. testing them locally.

**Comment by jawatts** [comment]
:+1: Looks good!

File: cfme/tests/pod/test_appliance_crud.py (hunk 7 above)
Comment (izapolsk) [reply cfme/tests/pod/test_appliance_crud.py]: it is already set. I just re-use openshift inventory config for cfme deployment. 
cfme inventory file also requires some data for correct deployment.

File: cfme/tests/pod/test_appliance_crud.py [hunk 8]
```diff
@@ -13,29 +19,158 @@
 ]
 
 
+ansible_config = """
```
Comment (jawatts) [inline cfme/tests/pod/test_appliance_crud.py]: You should filter your providers based on the 'podtesting' flag

**Pull request merged**